generator/vertx/src/main/java/io/fabric8/maven/generator/vertx/VertxGenerator.java

```diff
@@ -35,13 +52,98 @@
  */
 public class VertxGenerator extends JavaExecGenerator {
 
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private static final String VERTX_MAVEN_PLUGIN_GA = "io.fabric8:vertx-maven-plugin";
+    private static final String VERTX_CONFIG = "vertx.config";
+
+    private static final List<String> FALLBACK_PORTS = Arrays.asList("8080");
+
+    private static final String PORT_REGX = "([a-zA-Z0-9_]+)[\\.-_]port";
+    private static final Pattern PORT_PATTERN = Pattern.compile(PORT_REGX);
+
     public VertxGenerator(GeneratorContext context) {
         super(context, "vertx");
     }
 
     @Override
     public boolean isApplicable(List<ImageConfiguration> configs) throws MojoExecutionException {
-        // Disabled for now
+        return shouldAddImageConfiguration(configs)
+                && MavenUtil.hasPlugin(getProject(), VERTX_MAVEN_PLUGIN_GA);
+    }
+
+    @Override
+    protected boolean isFatJar() throws MojoExecutionException {
+        if (!hasMainClass() && isVertxPackage()) {
+            return true;
+        }
+        return super.isFatJar();
+    }
+
+
+    @Override
+    protected List<String> extractPorts() {
+        List<String> ports = portsFromConfig();
+        return ports != null ? ports : FALLBACK_PORTS;
+    }
+
+    protected List<String> portsFromConfig() {
+        List<String> answer = new ArrayList<>();
+        MavenProject project = getProject();
+        Plugin plugin = project.getPlugin(VERTX_MAVEN_PLUGIN_GA);
+        if (plugin != null) {
+            Xpp3Dom configuration = (Xpp3Dom) plugin.getConfiguration();
+            if (configuration == null) {
+                return null;
+            }
+            Xpp3Dom config = configuration.getChild("config");
+            String configPath = config != null ? config.getValue() : System.getProperty(VERTX_CONFIG, null);
```

Comment:
just my 2 cent, since maven plugin configuration can be at plugin level (global), goal level - it will be great if we can check both the options to make this a bit robust